File: app/src/main/java/com/zofers/zofers/staff/PreferenceService.java
package com.zofers.zofers.staff;

import android.content.Context;
import android.content.SharedPreferences;

public class PreferenceService {
    private Context context;

    public PreferenceService (Context c) {
        context = c;
    }

    public void setString (String key, String value) {
        getPreferences().edit().putString(key, value).apply();
    }

    public String getString (String key, String defValue) {
        return getPreferences().getString(key, defValue);
    }


    public SharedPreferences getPreferences(){
        return context.getSharedPreferences(context.getPackageName(), Context.MODE_PRIVATE);
    }

    public Context getContext() {
        return context;
    }
}

File: app/src/main/java/com/zofers/zofers/staff/PreferenceService.kt
package com.zofers.zofers.staff

import android.content.Context
import android.content.SharedPreferences

class PreferenceService(val context: Context) {

	val preferences: SharedPreferences
		get() = context.getSharedPreferences(context.packageName, Context.MODE_PRIVATE)

	fun setString(key: String, value: String?) {
		preferences.edit().putString(key, value).apply()
	}

	fun getString(key: String, defValue: String?): String? {
		return preferences.getString(key, defValue)
	}

	fun setBoolean(key: String, value: Boolean) {
		preferences.edit().putBoolean(key, value).apply()
	}

	fun getBoolean(key: String, defValue: Boolean): Boolean {
		return preferences.getBoolean(key, defValue)
	}
}
